### DISCUSSION TITLE
WPG 2.0 (WordPerfect Graphic 2.0) format support

### DISCUSSION THREAD

**Discussion by emendelson** [ImageMagick/ImageMagick#7085] [Development]
### ImageMagick version

7.1.1.27

### Operating system, version and so on

macOS 14

### Description

Is there any hope that ImageMagick will someday support the WordPerfect Graphic 2.0 standard, and not only WPG1? When I try to open a WPG 2.0 graphic, Magick can't convert it and produces an error message. Here is a sample image:

[YELOEYE.WPG.zip](https://github.com/ImageMagick/ImageMagick/files/14152507/YELOEYE.WPG.zip)

**Comment by fmw42** [comment]
On my Mac OSX Ventura with Imageamgick 7.1.1.27, I get

```
magick YELOEYE.WPG YELOEYE.png
magick: image file does not contain any image data `YELOEYE.WPG' @ error/wpg.c/ReadWPGImage/1632.
```

This should be noted on the formats page that it is only WPG 1 and not WPG 2 until that is fixed.

In the mean time you can get the shareware [GraphicConverter](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwiR5_Pg6Y-EAxUwEkQIHaIOD2IQFnoECAYQAQ&url=https%3A%2F%2Fwww.lemkesoft.de%2Fen%2Fproducts%2Fgraphicconverter%2F&usg=AOvVaw1XQgsEboOPz6Qo0h2WaX7u&opi=89978449). It will open that file.

**Comment by fmw42** [comment]
Here is a link that might help show the file format changes For WPG 2.  See for example https://www.fileformat.info/format/wpg/egff.htm

**Comment by emendelson** [comment]
Yes, thank you! I know that GraphicConverter will open the WPG2 format because I asked the developer to support it a few months ago. But, for many reasons, I hope that ImageMagick can support it also!

There seems to be no SDK for the 2.0 version of the format but the opensource `libpwg` has reverse-engineered it.

EDIT: And it took me ages to figure out that ImageMagick can write to the WPG1 format - the format page says only that it can read that format - it says `R`, not `RW` which would be correct.
